### PR TITLE
Update for newer terraform standards and resources.

### PR DESCRIPTION
**Pull request by dumptruckman**

### CODE REVIEW

File: s3.tf [hunk 1]
```diff
@@ -1,48 +1,73 @@
...
-    ]
 }
-EOF
 
+resource "aws_s3_bucket_policy" "web_policy" {
+  bucket = aws_s3_bucket.web.id
+  policy = data.aws_iam_policy_document.web_policy_document.json
 }
 
+data "aws_iam_policy_document" "web_policy_document" {
```
Comment (chris-malloy) [inline s3.tf:41]: love

**Review by chris-malloy — APPROVED** [review]

File: s3.tf (hunk 1 above)
Comment (kcchoate) [reply s3.tf:41]: Is this meant to be `data` and not a `resource`? I'm not sure how we would expect for this `aws_iam_policy_document` to already exist unless it is being generated by this terraform, and I'm not seeing where that would happen.

Comment (dumptruckman) [reply s3.tf:41]: From what I understand, this isn't creating a resource or pulling one from somewhere else. It simply represents data that can be used where needed.

Comment (kcchoate) [reply s3.tf:41]: Interesting. It was always my understanding that `data` blocks represented an existing piece of infrastructure defined outside of the terraform code in question. I'll read up on this, thank you!

**Pull request merged**